tests/testDipoleFitter.py

```diff
@@ -0,0 +1,407 @@
+
+        self.offsets = np_array([-2., 2.])  # pixel coord offsets between lobes of dipoles
+
+
+# First, test the algorithm itself (fitDipole()):
+# Create a simulated diffim (with dipoles) and a linear background gradient in the pre-sub images
+#   then compare the input fluxes/centroids with the fitted results.
+class DipoleFitAlgorithmTest(lsst_tests.TestCase):
+    """
+    A test case for dipole fit algorithm.
+
+    Test the dipole fitting algorithm on two dipoles within an image with
+    simulated noise.
+    """
+    def setUp(self):
+        self.params = DipoleFitTestGlobalParams()
+
+        offsets = self.params.offsets
+        self.dipole, (self.posImage, self.posCatalog), (self.negImage, self.negCatalog) = \
+            DipoleTestUtils.makeDipoleImage(
+                xcenPos=self.params.xc + offsets,
+                ycenPos=self.params.yc + offsets,
+                xcenNeg=self.params.xc - offsets,
+                ycenNeg=self.params.yc - offsets,
+                flux=self.params.flux, fluxNeg=self.params.flux,
+                gradientParams=self.params.gradientParams)
+
+        self.catalog = DipoleTestUtils.detectDipoleSources(self.dipole)
+
+    def tearDown(self):
+        del self.dipole, self.posImage, self.negImage
+        del self.catalog, self.posCatalog, self.negCatalog
+        del self.params
+
+    def testDipoleFitter(self):
+        """
+        Test the dipole fitting algorithm. Test that the resulting fluxes/centroids
+        are very close to the input values for both dipoles in the image.
+        """
+        if self.params.verbose:
+            for s in self.catalog:
+                fp = s.getFootprint()
+                print(fp.getBBox(), fp.getNpix())
+                for pk in fp.getPeaks():
+                    print('FOOTPRINT CENTER:', pk.getIy(), pk.getIx(), pk.getPeakValue())
+
+        offsets = self.params.offsets
+        for i, s in enumerate(self.catalog):
+            alg = DipoleFitAlgorithm(self.dipole, self.posImage, self.negImage)
+            result = alg.fitDipole(
+                s, rel_weight=1., separateNegParams=False,
+                verbose=self.params.verbose, display=self.params.display)
+
+            self.assertClose((result.psfFitPosFlux + abs(result.psfFitNegFlux))/2.,
+                             self.params.flux[i], rtol=0.02)
+            self.assertClose(result.psfFitPosCentroidX, self.params.xc[i] + offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitPosCentroidY, self.params.yc[i] + offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitNegCentroidX, self.params.xc[i] - offsets[i], rtol=0.01)
+            self.assertClose(result.psfFitNegCentroidY, self.params.yc[i] - offsets[i], rtol=0.01)
+
+
+# Test the task in the same way as the algorithm:
+# Also test that it correctly classifies the dipoles.
+class DipoleFitTaskTest(DipoleFitAlgorithmTest):
+    """ A test case for dipole fit task"""
+    def setUp(self):
```

Comment:
If this is a subclass of `DipoleFitAlgorithmTest` do we really need the `setUp` and `tearDown` methods to explicitly call the base class? I'm asking because I don't really know what `unittest` would do. If somehow the set up and tear downs won't happen automatically then can you add a comment to these methods indicating why they explicitly exist, otherwise delete them.